src/osm_configurator/view/toplevelframes/project_head_frame.py

```diff
@@ -32,8 +59,369 @@ def __init__(self, state_manager, export_controller, project_controller):
+
+    def _reduction_button_pressed(self):
+        # Buttons don't get activated or disabled, because if we change back into a state with the HeadFrame
+        # activate() will be called and to just that!
+
+        # Before changing a state the application auto saves!
+        self._save_project()
+
+        # Changing to Reduction State
+        if not self._state_manager.change_state(state_name_enum_i.StateName.REDUCTION):
+            alert_pop_up_i.AlertPopUp("Opening Reduction Failed!")
+
+    def _attractivity_button_pressed(self):
+        # Buttons don't get activated or disabled, because if we change back into a state with the HeadFrame
+        # activate() will be called and to just that!
+
+        # Before changing a state the application auto saves!
+        self._save_project()
+
+        # Changing to Attractivity_VIEW State
+        if not self._state_manager.change_state(state_name_enum_i.StateName.ATTRACTIVITY_VIEW):
+            alert_pop_up_i.AlertPopUp("Opening Attractivity Failed!")
+
+    def _aggregation_button_pressed(self):
+        # Buttons don't get activated or disabled, because if we change back into a state with the HeadFrame
+        # activate() will be called and to just that!
+
+        # Before changing a state the application auto saves!
+        self._save_project()
+
+        # Changing to Aggregation State
+        if not self._state_manager.change_state(state_name_enum_i.StateName.AGGREGATION):
+            alert_pop_up_i.AlertPopUp("Opening Aggregation Failed!")
+
+    def _calculate_button_pressed(self):
+        # Buttons don't get activated or disabled, because if we change back into a state with the HeadFrame
+        # activate() will be called and to just that!
+
+        # Before changing a state the application auto saves!
+        self._save_project()
+
+        # Changing to Calculate State
+        if not self._state_manager.change_state(state_name_enum_i.StateName.CALCULATION):
+            alert_pop_up_i.AlertPopUp("Opening Calculation Failed!")
+
+    def _options_button_pressed(self):
+        # Buttons don't get activated or disabled, because if we change back into a state with the HeadFrame
+        # activate() will be called and to just that!
+
+        # Before changing a state the application auto saves!
+        self._save_project()
+
+        # Changing to Option/Setting State
+        if not self._state_manager.change_state(state_name_enum_i.StateName.SETTINGS):
+            alert_pop_up_i.AlertPopUp("Opening Settings Failed!")
+
+    def _save_project(self):
+        # If the project can't be saved, an PopUp will pop up
+        if not self._project_controller.save_project():
+            alert_pop_up_i.AlertPopUp("Project could not be saved!")
+
+    def lock(self) -> bool:
+        if self._locked:
+            return False
+        else:
+            # Disabling all Buttons, except the save Button!
+            self._main_menu_button.configure(state="disabled",
```

Comment:
Please make sure everything is indented at the same level

Reply:
with now a list for buttons, all buttons get disabled and the save button enabled afterwards, making sure all buttons are disabled except for the save buttons, as intended